test/char_array.correct.txt

```diff
@@ -0,0 +1,50 @@
+Running test: char_array_eq
+PASS
+Running test: char_array_uneq
+FAIL
+Running test: string_eq
+FAIL
+Running test: string_uneq1
+FAIL
+Running test: string_uneq2
+FAIL
+Running test: string_uneq3
+FAIL
+Running test: string_uneq4
+FAIL
+Running test: string_uneq5
+FAIL
+
+*** Results ***
+** Test case "char_array_eq": PASS
+** Test case "char_array_uneq": FAIL
+In test , line 48: 
+{ s, p, a, m } != { e, g, g }
+
+** Test case "string_eq": FAIL
+In test , line 13: 
+spam != spam
```

Comment:
This strikes me as a bit odd. I understand why the failure happens, but then this output doesn't give any indication as to what the difference is.

The old output of `{ s, p, a, m, }` is only slightly better because of the extra comma. 

I can think of a couple possible solutions, but they seem like more effort to make something that's inherently messy a little less messy. What do you think?

Reply:
Suggestion:
```
@@ -367,6 +367,17 @@ void assert_sequence_equal(First&& first, Second&& second, int line_number) {
     auto it2 = begin(second);
     auto end1 = end(first);
     auto end2 = end(second);
+    auto len1 = std::distance(it1, end1);
+    auto len2 = std::distance(it2, end2);
+    if (len1 != len2) {  // different number of elements
+        std::ostringstream reason;
+        print(reason, first);
+        reason << " (size=" << len1 << ") != ";
+        print(reason, second);
+        reason << " (size=" << len2 << ")";
+        throw TestFailure(reason.str(), line_number);
+    }
+
```

Then the error message would be:
```
** Test case "string_eq": FAIL
In test , line 13: 
spam (size=5) != spam (size=4)
```

Thoughts?

Reply:
Yes, I think that provides enough information for someone reasonably familiar with c-strings to figure it out.

Reply:
Slight modification to the following:
```
spam != spam (sizes differ: 5 != 4)
```

Also, I modified `assert_sequence_equal` so that it prints out the position and elements that differ:
```
{ s, p, a, m } != { e, g, g, s } (elements at position 0 differ: s != e)
```

Reply:
Looks good to me!